app/controllers/admin/dashboard_controller.rb

```diff
@@ -0,0 +1,57 @@
+class Admin::DashboardController < ApplicationController
+  before_action :authorize_admin
+
+  def index
+    pending_orders = Order
```

Comment:
You have a lot of lines here that repeat themselves. You should extract them, like:
```ruby
orders = Order.where.not(status: :cart) # and all other lines that also repeat
# and then do
completed_orders = orders.where(status: :pending)
completed_orders = orders.where(status: :completed)
# etc.
```